crates/runtime/tests/models/openai.rs

```diff
@@ -262,6 +268,134 @@ async fn openai_test_embeddings() -> Result<(), anyhow::Error> {
+    });
+
+    tokio::select! {
+        () = tokio::time::sleep(std::time::Duration::from_secs(10)) => {
+            return Err(anyhow::anyhow!("Timed out waiting for components to load"));
+        }
+        () = rt.load_components() => {}
+    }
+
+    runtime_ready_check(&rt).await;
+
+    let response = send_chat_completions_request(
+        http_base_url.as_str(),
+        vec![
+            ("system".to_string(), "You are an assistant that responds to queries by providing only the requested data values without extra explanation.".to_string()),
+            ("user".to_string(), "Provide the total number of records in the taxi trips dataset. If known, return a single numeric value.".to_string()),
+        ],
+        "openai_model",
+        false,
+    ).await?;
+
+    insta::assert_snapshot!(
+        "chat_completion",
+        normalize_chat_completion_response(response, false)
+    );
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn openai_test_chat_messages() -> Result<(), anyhow::Error> {
+    let _tracing = init_tracing(None);
+
+    verify_env_secret_exists("SPICE_OPENAI_API_KEY")
+        .await
+        .map_err(anyhow::Error::msg)?;
+
+    let app = AppBuilder::new("text-to-sql")
+        .with_dataset(get_taxi_trips_dataset())
+        .build();
+
+    let rt = Arc::new(Runtime::builder().with_app(app).build().await);
+
+    let _tracing = init_tracing_with_task_history(None, &rt);
+
+    tokio::select! {
+        () = tokio::time::sleep(std::time::Duration::from_secs(30)) => {
+            return Err(anyhow::anyhow!("Timed out waiting for components to load"));
+        }
+        () = rt.load_components() => {}
+    }
+
+    runtime_ready_check(&rt).await;
+
+    let mut model_with_tools = get_openai_model("gpt-4o-mini", "openai_model");
+    model_with_tools
+        .params
+        .insert("spice_tools".to_string(), "auto".into());
+
+    let model_secrets = get_params_with_secrets(&model_with_tools.params, &rt).await;
+
+    let tool_model = try_to_chat_model(&model_with_tools, &model_secrets, Arc::clone(&rt)).await?;
+
+    let req = CreateChatCompletionRequestArgs::default()
+                .messages(vec![ChatCompletionRequestSystemMessageArgs::default()
+                    .content("You are an assistant that responds to queries by providing only the requested data values without extra explanation.".to_string())
+                    .build()?
+                    .into(),ChatCompletionRequestUserMessageArgs::default()
+                    .content("Provide the total number of records in the taxi trips dataset. If known, return a single numeric value.".to_string())
+                    .build()?
+                    .into()])
+                .build()?;
+
+    let task_start_time = std::time::SystemTime::now();
+    let response = tool_model.chat_request(req).await?;
+
+    insta::assert_snapshot!("chat_1_choices", format!("{:?}", response.choices));
```

Comment:
Is the possibility of hallucination/different response values going to cause problems with `insta`?

Reply:
This is possible but the probability should be very low, I'll be checking this to see how stable it is and improve if needed, so far it worked well for me.